In [10]:
from lenskit.algorithms import als
from lenskit import batch
from lenskit.metrics.predict import rmse
import joblib
import pandas as pd


df: pd.DataFrame = joblib.load('test-gt16-s15.joblib').drop_duplicates()

X_val = df.drop(columns=['rating'])
y_val = df['rating']

model: als.BiasedMF = joblib.load('model-gt16-s85.joblib')
pred = batch.predict(model, X_val)

# lenskit messed up the index but keeps the order, so we can fix it like this:
pred.index = X_val.index

rmse(pred['prediction'], y_val) # 0.9773730468960676

0.9630560905111525

#### Validation results:
##### BiasedMF:
- 80/20 split with data after 2018: RMSE = 0.9773730468960676
- 85/15 split with data after 2016: RMSE = 0.9630560905111525